src/sleep/sleep.c

```diff
@@ -39,18 +39,19 @@ STATIC_DESTRUCTOR_REGISTER(arg_verb, freep);
 
 static int write_hibernate_location_info(const HibernateLocation *hibernate_location) {
         char offset_str[DECIMAL_STR_MAX(uint64_t)];
+        char resume_str[DECIMAL_STR_MAX(dev_t)];
         int r;
 
         assert(hibernate_location);
         assert(hibernate_location->swap);
-        assert(hibernate_location->resume);
 
-        r = write_string_file("/sys/power/resume", hibernate_location->resume, WRITE_STRING_FILE_DISABLE_BUFFER);
+        xsprintf(resume_str, "%u:%u", major(hibernate_location->resume), minor(hibernate_location->resume));
+        r = write_string_file("/sys/power/resume", resume_str, WRITE_STRING_FILE_DISABLE_BUFFER);
```

Comment:
btw, there's write_string_filef() which allows doing these two lines in one

Reply:
I'm using `resume_str` in the `log_debug` message a few lines down so I'd have to `xsprtintf` the string anyway. Would it still be desirable to use the `write_string_filef` and then `xsprintf` for the message?